tuf/tuf.go

```diff
@@ -49,6 +49,20 @@ func (err ErrNotLoaded) Error() string {
 	return fmt.Sprintf("%s role has not been loaded", err.role)
 }
 
+// ErrStopWalk - used by visitor functions to signal WalkTargets to stop walking
+type ErrStopWalk struct{}
```

Comment:
This may not be necessary, since this PR as is will also stop walking on returning `nil`.  When @endophage and I discussed the design of the walker it seemed useful to have different states for "finishing" a walk and "aborting" a walk but the use case hasn't presented itself so far.  I can delete this or leave this in, depending on what we think.